geopy/geocoders/herev7.py

```diff
@@ -0,0 +1,286 @@
+import collections.abc
+import warnings
+from functools import partial
+from urllib.parse import urlencode
+
+from geopy.exc import (
+    ConfigurationError,
+    GeocoderAuthenticationFailure,
+    GeocoderInsufficientPrivileges,
+    GeocoderQuotaExceeded,
+    GeocoderServiceError,
+    GeocoderUnavailable,
+)
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import join_filter, logger
+
+__all__ = ("HereV7", )
+
+
+class HereV7(Geocoder):
+    """Geocoder using the HERE Geocoder API.
+
+    Documentation at:
+        https://developer.here.com/documentation/geocoding-search-api/
+    """
+
+    structured_query_params = {
+        'street',
+        'houseNumber',
+        'postalCode',
+        'city',
+        'district',
+        'county',
+        'state',
+        'country'
+    }
+
+    geocode_path = '/v1/geocode'
+    reverse_path = '/v1/revgeocode'
+
+    def __init__(
+            self,
+            *,
+            apiKey=None,
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            ssl_context=DEFAULT_SENTINEL,
+            adapter_factory=None
+    ):
+        """
+
+        :param str apiKey: Should be a valid HERE Maps apiKey.
+            More authentication details are available at
+            https://developer.here.com/authenticationpage.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+        :param callable adapter_factory:
+            See :attr:`geopy.geocoders.options.default_adapter_factory`.
+
+            .. versionadded:: 2.0
```

Comment:
This `versionadded` directive is not needed for that param, because the whole HereV7 class would be added later than the already released 2.0.

Reply:
Fixed in 79abd5d